### DISCUSSION TITLE
What is the NodeHtmlBlock block_type field, please?

### DISCUSSION THREAD

**Discussion by mightyiam** [kivikakk/comrak#351] [Q&A] [answered]
Sorry, I couldn't figure out what this field represents.

https://github.com/kivikakk/comrak/blob/4c909b654cfd3600ab3e96c7186228ba7c8f79cd/src/nodes.rs#L337-L346

**Comment by kivikakk** [comment, marked as answer]
Haha, good call, it's not very clear, is it? Here's where it's assigned in the code:

https://github.com/kivikakk/comrak/blob/4c909b654cfd3600ab3e96c7186228ba7c8f79cd/src/parser/mod.rs#L1052-L1065

This is also not very clear — tl;dr it stores a number from 1 to 7, corresponding to these seven start/end condition pairs in [CommonMark § 4.6](https://spec.commonmark.org/0.30/#html-blocks):

> 1.  **Start condition:**  line begins with the string `<pre`, `<script`, `<style`, or `<textarea` (case-insensitive), followed by a space, a tab, the string `>`, or the end of the line.\
> **End condition:**  line contains an end tag `</pre>`, `</script>`, `</style>`, or `</textarea>` (case-insensitive; it need not match the start tag).
> 
> 2.  **Start condition:** line begins with the string `<!--`.\
> **End condition:**  line contains the string `-->`.
> 
> 3.  **Start condition:** line begins with the string `<?`.\
> **End condition:** line contains the string `?>`.
> 
> 4.  **Start condition:** line begins with the string `<!` followed by an ASCII letter.\
> **End condition:** line contains the character `>`.
> 
> 5.  **Start condition:**  line begins with the string `<![CDATA[`.\
> **End condition:** line contains the string `]]>`.
> 
> 6.  **Start condition:** line begins the string `<` or `</` followed by one of the strings (case-insensitive) `address`, `article`, `aside`, `base`, `basefont`, `blockquote`, `body`, `caption`, `center`, `col`, `colgroup`, `dd`, `details`, `dialog`, `dir`, `div`, `dl`, `dt`, `fieldset`, `figcaption`, `figure`, `footer`, `form`, `frame`, `frameset`, `h1`, `h2`, `h3`, `h4`, `h5`, `h6`, `head`, `header`, `hr`, `html`, `iframe`, `legend`, `li`, `link`, `main`, `menu`, `menuitem`, `nav`, `noframes`, `ol`, `optgroup`, `option`, `p`, `param`, `section`, `source`, `summary`, `table`, `tbody`, `td`, `tfoot`, `th`, `thead`, `title`, `tr`, `track`, `ul`, followed by a space, a tab, the end of the line, the string `>`, or the string `/>`.\
> **End condition:** line is followed by a blank line.
> 
> 7.  **Start condition:**  line begins with a complete open tag (with any tag name other than `pre`, `script`, `style`, or `textarea`) or a complete closing tag, followed by zero or more spaces and tabs, followed by the end of the line.\
> **End condition:** line is followed by a blank line.

**Reply by mightyiam** [reply]
Thank you!
